app/src/main/java/cm/aptoide/pt/app/view/AppViewPresenter.java

```diff
@@ -897,6 +896,7 @@ private void handleClickOnRetry() {
             .getKeywords())
         .flatMap(similarAppsViewModel -> appViewManager.shouldLoadNativeAds()
             .doOnSuccess(similarAppsViewModel::setShouldLoadNativeAds)
+            .doOnSuccess(__ -> similarAppsViewModel.setFromMatureApp(appViewModel.isMature()))
```

Comment:
Why are we updating this flag after object is instantiated when we should inject this info?

Reply:
Stupidly I did not inject it because this model was also used for appcoins and there are no ads in the appcoins similar apps bundle, so at the time I thought I should not inject it as it would not depend on it.
Refactored it and also changed the should load native ads flag.